class/actions_discountrules.class.php

```diff
@@ -298,7 +298,7 @@ public function addMoreActionsButtons($parameters, &$object, &$action, $hookmana
 
 				$params = array(
 						'attr' => array(
-								'data-document-url' => $_REQUEST['PHP_SELF'] . '?id=' . $object->id . '&token=' . newToken(),
+								'data-document-url' => $_REQUEST['PHP_SELF'] . '?id=' . $object->id . '&token=' . $newToken,
```

Comment:
Discount rule n'est pas compatible Dolibarr < 12 Donc il faut directement utiliser newToken()

```suggestion
								'data-document-url' => $_REQUEST['PHP_SELF'] . '?id=' . $object->id . '&token=' . newToken(),
```